app/features/history/camera/HistoryVideoRecorderActions.js

```diff
@@ -15,17 +16,63 @@ export const stopRecording = () => ({
     type: STOP_RECORDING_HISTORY
 });
 
-export const dismissRecording = () => {
+export const dismissRecording = () => (dispatch, getState) => {
+    var state = getState();
+
+    cancelVideoAnalytics(state);
+
     Analytics.setCurrentScreen('history');
     
-    return {
+    dispatch({
         type: DISMISS_HISTORY_VIDEO_RECORDER
-    }
+    });
+};
+
+export const saveVideo = (setID, videoFileURL, videoType) => (dispatch, getState) => {
+    var state = getState();
+
+    saveVideoAnalytics(state);
+
+    dispatch({
+        type: SAVE_HISTORY_VIDEO,
+        setID: setID,
+        videoFileURL: videoFileURL,
+        videoType: videoType
+    });
+};
+
+export const saveVideoError = (setID) => (dispatch, getState) => {
+    var state = getState();
+
+    saveVideoErrorAnalytics(state);
+
+    dispatch({
+        type: SAVE_VIDEO_ERROR,
+    });
+}
+
+// TODO: check that the duration calculation operates properly
```

Comment:
Check that the new DurationsSelectors.getHistoryVideoRecorderDuration is working properly

Reply:
done

Reply:
Are you sure? This is a new function that you didn't write

Reply:
Yes